File: src/backend/src/main/java/freiraumbande/members/service/MemberService.java
package freiraumbande.members.service;

import freiraumbande.members.dto.MemberDTO;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.util.List;

public interface MemberService {
    List<MemberDTO> findAll();

    /** Legt ein Mitglied an — das Foto ist optional. */
    MemberDTO create(String name, String role, MultipartFile photo) throws IOException;

    /**
     * Aktualisiert Name/Rolle. Ein neues Foto ist optional (null = behalten);
     * removePhoto entfernt das vorhandene Foto (wird von einem neuen Foto übersteuert).
     */
    MemberDTO update(Long id, String name, String role, MultipartFile photo, boolean removePhoto)
            throws IOException;

    void delete(Long id);
}

File: src/frontend/src/services/memberService.ts
import api from './api'
import { Member } from '../types'

export interface MemberFormData {
  name: string
  role?: string
  /** Optional — beim Bearbeiten heißt leer: Foto behalten. */
  file?: File
  /** Entfernt beim Bearbeiten das vorhandene Foto (wird von file übersteuert). */
  removePhoto?: boolean
}

function toFormData({ name, role, file, removePhoto }: MemberFormData): FormData {
  const formData = new FormData()
  formData.append('name', name)
  if (role) formData.append('role', role)
  if (file) formData.append('file', file)
  if (removePhoto) formData.append('removePhoto', 'true')
  return formData
}

export const memberService = {
  getAll: async (): Promise<Member[]> => {
    const { data } = await api.get<Member[]>('/api/members')
    return data
  },

  create: async (member: MemberFormData): Promise<Member> => {
    const { data } = await api.post<Member>('/api/members', toFormData(member), {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
    return data
  },

  update: async (id: number, member: MemberFormData): Promise<Member> => {
    const { data } = await api.put<Member>(`/api/members/${id}`, toFormData(member), {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
    return data
  },

  delete: async (id: number): Promise<void> => {
    await api.delete(`/api/members/${id}`)
  },
}
